components/ui/CopyButton.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import { useState } from "react";
+import { BoxButton } from "./BoxButton";
+import CopyIcon from "public/assets/icons/copy.svg";
+import CopyActiveIcon from "public/assets/icons/copy_active.svg";
+
+export const CopyButton = ({ content, children, compact, classes }: {content: string, children?: any, compact?: boolean, classes?: string }) => {
+    const [copied, setCopied] = useState(false);
+
+    function copy() {
+        const el = document.createElement('textarea');
+        el.value = content;
+        document.body.appendChild(el);
+        el.select();
+        document.execCommand('copy');
+        document.body.removeChild(el);
+
+        setCopied(true);
+        setTimeout(() => setCopied(false), compact ? 800 : 2000);
+    }
+
+    if (compact) {
+        return (
+            <div className={`${classes} cursor-pointer`} onClick={copy}>
```

Comment:
```suggestion
            <button className={`${classes} cursor-pointer`} onClick={copy}>
```
Should be a `button` so the browser and screen readers know what this is. Since the content is only an icon, the button should have a `title` tag to describe its functionality ("Copy", then "Copied" on click).